src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1284,18 +1284,25 @@ protected override void Validate(object arguments, EngineIntrinsics engineIntrin
                     null, Metadata.ValidateCountNotInArray);
             }
 
+            if (MinLength == MaxLength && len != MaxLength)
```

Comment:
@iSazonov: @lzybkr will have to confirm, but here's my understanding:

* For the `MinLength != MaxLength` case: Use the just-added `The parameter requires at least {0} value(s) and no more than {1} value(s) - {2} value(s) were provided.` message.

* For the `MinLength == MaxLength` case: add a new message along the lines of `The parameter requires exactly {0} value(s)  - {2} value(s) were provided.`